docs/source/jupyterlab-requirements-ui.md

```diff
@@ -22,23 +22,35 @@ Initially, no dependencies are identified if you start a new notebook as metadat
 The extension checks in the notebook metadata in order to identify them every time you restart a notebook.
 Moreover it verifies that the kernel you are using is matching your dependencies. If not it warns to use install button again to avoid weird behaviours.
 
-You can start adding your packages using the central add button and once you select package name and version, remember to add your package using add button in action,
-otherwise it won't be saved (in the future this behaviour will not be necessary due to the autocompletion feature):
+You can start adding your packages using the central add button, which will open a new package row to edit.
+1. First type out a pakcage. For packages that are avialable in PyPi, an informational popup will be shown to indciate that it is a valid package.
+2. Then choose a constraint. If the package is both valid and known by Thoth, clicking the dialog box will open a dropdown (explained in detail bellow). If the package is not known by Thoth, then you will have to manually type the constraints.
```

Comment:
I am checking they are on PyPI and then to get the versions I am checking Thoth. Because Thoth doesn't have everything PyPI has. So checking Thoth knowledge graph

Reply:
> I am checking they are on PyPI and then to get the versions I am checking Thoth. Because Thoth doesn't have everything PyPI has. So checking Thoth knowledge graph

But this should happen only if the thoth resolution is used right? otherwise it should check PyPI if pipenv is selected. 

can you open an issue to fix this, please?

Reply:
The issue is that in the current UI, The option to select the resolution engine doesn't appear until after you have set all your packages and versions.

Reply:
I think the question at hand is a bit fuzzy, and may require extra indicators to be 100% clear.

To explain a bit more clearly:

Using PyPI data, I can get both the package summary and versions. Currently I use the PyPI data for the summary and Thoth data for the versions. This becomes an issue if Thoth doesn't have that package in its knowledge graph.

I dont know if the user wants to use Thoth resolution engine, so like you said, we will default to PyPI. However If the user then chooses to use Thoth in the options form, but previously selected a package not known by Thoth, then there is an issue.

In the options form, I can do a check for each package selected to see if it is in the knowledge graph. If one is not, then I give a warning and link them to the issue template.

We can discuss it in the tech talk as well to make sure we are on the same page with things.